packages/core/src/external/carequality/ihe-gateway-v2/inbound/xcpd/create/xcpd-response.ts

```diff
@@ -0,0 +1,294 @@
+  const patientResource = isSuccessfulPatientDiscoveryResponse(response)
+    ? response.patientResource
+    : undefined;
+  if (!patientResource) {
+    throw new Error("patientResource is missing in the response");
+  }
+  if (!externalGatewayPatient) {
+    throw new Error("externalGatewayPatient is missing in the response");
+  }
+  const subject = {
+    "@_typeCode": "SBJ",
+    "@_contextConductionInd": "false",
+    registrationEvent: {
+      "@_classCode": "REG",
+      "@_moodCode": "EVN",
+      statusCode: {
+        "@_code": "active",
+      },
+      subject1: {
+        "@_typeCode": "SBJ",
+        patient: {
+          "@_classCode": "PAT",
+          id: {
+            "@_extension": externalGatewayPatient.id,
+            "@_root": externalGatewayPatient.system,
+          },
+          statusCode: "active",
+          patientPerson: {
+            "@_classCode": "PSN",
+            "@_determinerCode": "INSTANCE",
+            name: patientResource.name.map(n => ({
+              family: n.family,
+              ...n.given?.reduce((acc: { [key: string]: string }, givenName: string) => {
+                acc.given = givenName;
+                return acc;
+              }, {}),
+            })),
+            telecom: patientResource.telecom?.map(t => ({
+              "@_use": t.system,
+              "@_value": t.value,
+            })),
+            administrativeGenderCode: {
+              "@_code": mapFhirToMetriportGender(patientResource.gender),
+            },
+            birthTime: {
+              "@_value": patientResource.birthDate,
+            },
+            addr: patientResource.address?.map(a => ({
+              streetAddressLine: a.line?.join(", "),
+              city: a.city,
+              state: a.state,
+              postalCode: a.postalCode,
+              country: a.country,
+            })),
+          },
+        },
+      },
+    },
+  };
+  return subject;
+}
+
+function createAckAndQueryResponseCode(response: InboundPatientDiscoveryResp): {
+  ack: string;
+  queryResponseCode: string;
+} {
+  const queryResponseCode =
+    response.patientMatch === true
+      ? queryResponseCodes.OK
+      : response.patientMatch === false
+      ? queryResponseCodes.NF
+      : queryResponseCodes.AE;
+  const ack =
+    response.patientMatch === true || response.patientMatch === false ? ackCodes.AA : ackCodes.AE;
+  return { ack, queryResponseCode };
+}
+
+function createIti55SoapBody(
+  request: InboundPatientDiscoveryReq,
+  response: InboundPatientDiscoveryResp
+): object {
+  const { ack, queryResponseCode } = createAckAndQueryResponseCode(response);
+  const queryByParameter = createQueryByParameter(request);
+  const subject = response.patientMatch ? createSubjectAndRegistrationEvent(response) : undefined;
+
+  const soapBody = {
+    "@_xmlns:urn": namespaces.hl7,
+    PRPA_IN201306UV02: {
+      id: {
+        "@_root": uuidv7(), // TODO change this if we ever end up keeping track of these.
```

Comment:
ditto: how to keep track of this item after this PR is merged.

nit: why the diff approach to UUID generation, v4 vs v7 in diff places of the PR?

Reply:
changed to v4